onnxruntime/core/providers/cpu/nn/conv_attributes.h

```diff
@@ -118,20 +118,133 @@ struct ConvAttributes {
           rank + dim >= pads_p.size()) {
         return ORT_MAKE_STATUS(ONNXRUNTIME, FAIL, "Out of bound access to array");
       }
-      int64_t dim_size = 0;
+      int64_t output_dim_size = 0;
       ORT_RETURN_IF_ERROR(ComputePadAndOutputShape(input_shape[dim],
                                                    strides_p[dim],
                                                    kernel_shape[dim],
                                                    dilations_p[dim],
                                                    auto_pad,
                                                    pads_p.at(dim),
                                                    pads_p.at(input_shape.NumDimensions() + dim),
-                                                   dim_size,
+                                                   output_dim_size,
                                                    force_symmetric_auto_padding));
-      if (dim_size <= 0) {
+      if (output_dim_size <= 0) {
         return Status(common::ONNXRUNTIME, common::INVALID_ARGUMENT, "Invalid input shape: " + input_shape.ToString());
       }
-      output_shape.push_back(dim_size);
+      output_shape.push_back(output_dim_size);
+    }
+    return Status::OK();
+  }
+
+  // Use this method when pads are to be made symmetrical (if they are asymmetric)
+  // and to collect metadata regarding the portion of the output (with "adjusted" pads)
+  // to be sliced off to make the output correspond to the "actual" asymmetric paddings
+  Status InferOutputShapeWithAdjustedPads(const TensorShape& input_shape,
+                                          const std::vector<int64_t>& kernel_shape,
+                                          const std::vector<int64_t>& strides_p,
+                                          const std::vector<int64_t>& dilations_p,
+                                          std::vector<int64_t>& pads_p,
+                                          std::vector<int64_t>& output_shape,
+                                          std::vector<int64_t>& output_shape_with_revised_pads,
+                                          bool& post_slicing_needed,
+                                          std::vector<int64_t>& slice_starts,
+                                          std::vector<int64_t>& slice_ends,
+                                          std::vector<int64_t>& slice_axes) const {
+    size_t rank = input_shape.NumDimensions();
+    for (size_t dim = 0; dim < rank; ++dim) {
+      if (dim >= strides_p.size() || dim >= kernel_shape.size() ||
+          dim >= dilations_p.size() || dim >= pads_p.size() ||
+          rank + dim >= pads_p.size()) {
+        return ORT_MAKE_STATUS(ONNXRUNTIME, FAIL, "Out of bound access to array");
+      }
+      int64_t output_dim_size = 0;
```

Comment:
nit: could check this prior to the loop.

Reply:
sorry - didn't get this. which check can be done prior to the loop ?

Reply:
sorry - codeflow got lost with the lines involved. You can check for potential out-of-bounds access once prior to the 'for' loop.

Reply:
done, thanks